libs/driver/src/VideoDriver.cpp

```diff
@@ -72,5 +75,54 @@ VideoMode VideoDriver::FindClosestVideoMode(const VideoMode& mode) const
 void VideoDriver::SetNewSize(VideoMode windowSize, Extent renderSize)
 {
     windowSize_ = windowSize;
-    renderSize_ = renderSize;
+    screenRenderSize_ = renderSize;
+    viewRenderSize_ = guiScale_.screenToView<Extent>(renderSize);
+
+    const auto ratioXY = PointF(screenRenderSize_) / PointF(windowSize_.width, windowSize_.height);
+    dpiScale_ = (ratioXY.x + ratioXY.y) * .5f; // use the average ratio of both axes
```

Comment:
Similar: Don't transform division by hand:
```suggestion
    dpiScale_ = (ratioXY.x + ratioXY.y) / 2f; // use the average ratio of both axes
```